File: src/main/java/com/projetopicpay/springao/services/AutorizadorService.java
package com.projetopicpay.springao.services;

import com.projetopicpay.springao.dtos.AutorizadorResponse;
import com.projetopicpay.springao.exception.PagamentoNaoAutorizado;
import org.springframework.http.HttpStatusCode;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestClient;

@Service
public class AutorizadorService {

    public Boolean autorizar(){
        AutorizadorResponse res = RestClient.create()
                .get()
                .uri("https://util.devi.tools/api/v2/authorize")
                .retrieve()
                .onStatus(HttpStatusCode::is4xxClientError, (request, response) -> {
                    throw new PagamentoNaoAutorizado("Servico externo de pagamento negou a transacao");
                })
                .body(AutorizadorResponse.class);

        return res != null && res.data().authorization();
    }

}

File: src/services/autorizadorService.ts
import axios from 'axios';
import { PagamentoNaoAutorizadoException } from '../exceptions/AppError';

interface AutorizadorResponse {
  status: string;
  data: {
    authorization: boolean;
  };
}

export async function autorizar(): Promise<void> {
  const { data } = await axios.get<AutorizadorResponse>(
    'https://util.devi.tools/api/v2/authorize'
  );

  if (!data.data.authorization) {
    throw new PagamentoNaoAutorizadoException();
  }
}
